results-processor/wptreport.py

```diff
@@ -509,10 +529,12 @@ def create_test_run(report, labels_str, uploader, secret,
     labels = prepare_labels(report, labels_str, uploader)
     assert len(labels) > 0
 
+    labels |= normalize_product(report)
```

Comment:
I'd prefer calling this inside `prepare_labels` (or perhaps even inline it there).

Reply:
This means `prepare_labels` can change the name of the product within the report, which seems an even more surprising outcome.

Reply:
Ahh, I didn't see `normalize_product` would also modify `report.run_info`. What about not doing that?

In other words, instead of "normalizing the product name" in the report, think of it as adding a special extra label. That way, it's more similar to `prepare_labels` than `normalize_version` (which is a hack that I don't like much).

Reply:
Doing what exactly? We need these runs to show up as "Edge" runs in the UI (that it's using a different runner should probably be considered an implementation detail), and likely want Chrome/CDP runs to show up as "Chrome" in the UI.

If we add a magic label, we'll need to encode some way of expressing "this is a product name override", and ensuring everywhere which reads the product name reads from there.

It seems very much easier to change the actual product name in the report, because then we don't need to deal with some override. That said, given we end up with the report labelled with both "edge" and "edge_webdriver", we might want to clarify which label is the original product name (though it is fairly obvious to a human!).

Reply:
Right, I was thinking not modifying `report.run_info`, in which case we'd need to override the product name in the request param to `/api/resuls/create`, which would be convoluted...

I was a bit unhappy about the fact that two `normalize_*` functions ended up in different places (in and out of the class) with slightly different behaviours, but I guess it's acceptable.

Reply:
I do agree it's not ideal, but I'm not sure what the best solution here really is. All the options seem not great, and at least this isn't a large change now…